content/concepts/configuration.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Configuration
+---
+
+## Overview 
+You may have noticed that few webpack configurations look exactly alike. This is because **webpack's configuration file is a JavaScript file that exports an object.** This object, is then parsed by webpack based upon its defined properties.
+
+### Examples
+The following examples below describe how webpacks configuration object can be both expressive and configurable because _it is code_: 
+
+#### The simplest of examples
+**webpack.config.js**
+```javascript 
+module.exports = {
+  entry: './foo.js',
+  output: {
+    path: 'dist',
+    filename: 'foo.bundle.js'
+  }
+}
+
+```
+
+#### Generating multiple targets and configs and exporting the entire array.
+**webpack.config.js**
+```javascript
+var path = require('path');
+var webpack = require('webpack');
+var webpackMerge = require('webpack-merge');
+
+var baseConfig = {  
+  target: 'async-node',
+  entry: {
+    entry: './entry.js'
+  },
+  output: {
+    filename: '[name].js',
+    path: path.resolve(__dirname, './dist')
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'inline',
+      filename: 'inline.js',
+      minChunks: Infinity
+    }),
+    new webpack.optimize.AggressiveSplittingPlugin({
+        minSize: 5000,
+        maxSize: 10000
+    }),
+  ]
+};
+
+let targets = ['web', 'webworker', 'node', 'async-node', 'node-webkit', 'electron-main'].map((target) => {
```

Comment:
It's simple enough to extract the code fragments using something like `marked`. Given we use ellipsis here and there the code won't be valid always by definition. That said, if you want to give it a go, I can point you to the right direction. The basic idea goes like this:

1. Glob all Markdown files.
2. Hook into marked lexer and extract code fragments. [Example](https://github.com/webpack/webpack.io/blob/develop/utilities/markdown.js).
3. Eliminate any custom markup (i.e., ellipsis).
4. Run eslint against the code fragments.

I set up #92 for this task.